File: src/main/webapp/WEB-INF/static/core/service/CampeonatoService.js
/**
 * 
 */
app.factory('CampeonatoService', ['$http', function($http) {
	
	var serverURL = function(url) {
		return SERVER_APP + '/campeonato' + url;
	};
	
	/**
	 * pega campeonato pelo id
	 */
	var _get = function(id) {
		return $http.get(serverURL('/get/'+id));
	};
	
	var _save = function(campeonato, isUsuario) {
		return $http.post(SERVER_APP + '/campeonato/save/usuario/'+isUsuario, campeonato);
	};
	
	/**
	 * Busca campeonatos para o perfil desse usuario
	 */
	var _getListByUsuario = function(usuario) {
		return $http.get(serverURL('/get/list/por/usuario/'+usuario.id));
	};
	
	/**
	 * Busca campeonatos para o perfil desse usuario
	 */
	var _getListByUsuarioAdm = function(usuario) {
		return $http.get(serverURL('/get/list/by/usuario/adm/'+usuario.id));
	};
	
	return {
		
		get : _get,
		
		save : _save,
		
		getListByUsuario : _getListByUsuario,
		
		getListByUsuarioAdm : _getListByUsuarioAdm
		
	};
	
}]);

File: src/main/java/com/diegolirio/jcampeonato/service/CampeonatoService.java
package com.diegolirio.jcampeonato.service;


import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.diegolirio.jcampeonato.dao.CampeonatoDao;
import com.diegolirio.jcampeonato.dao.UsuarioDao;
import com.diegolirio.jcampeonato.dao.UsuarioPerfilCampeonatoDao;
import com.diegolirio.jcampeonato.model.Campeonato;
import com.diegolirio.jcampeonato.model.Perfil;
import com.diegolirio.jcampeonato.model.Usuario;
import com.diegolirio.jcampeonato.model.UsuarioPerfilCampeonato;

@Service("campeonatoService")
public class CampeonatoService extends AbstractGenericService<Campeonato> {

	@Autowired
	private CampeonatoDao campeonatoDao;

	@Autowired
	private UsuarioDao usuarioDao;

	@Autowired
	private UsuarioPerfilCampeonatoDao usuarioPerfilCampeonatoDao;

	public List<Campeonato> getListPorUsuario(Usuario usuario) {
		return this.campeonatoDao.getListPorUsuario(usuario);
	}

	public List<Campeonato> getListByUsuarioAdm(Usuario usuario) {
		return this.campeonatoDao.getListByUsuarioAdm(usuario);
	}

	
	/**
	 * Grava campeoanto, se for um novo insere perfil como admistrador
	 * @param campeonato
	 * @param usuarioId
	 */
	public void save(Campeonato campeonato, long usuarioId) {
		super.save(campeonato);
		if(campeonato.getId() == 0) {
			// busca usuario
			Usuario usuario = this.usuarioDao.get(Usuario.class, usuarioId);
			// criar o perfil como adm
			UsuarioPerfilCampeonato usuPerfilCamp = new UsuarioPerfilCampeonato();			
			usuPerfilCamp.setCampeonato(campeonato);
			usuPerfilCamp.setUsuario(usuario);
			usuPerfilCamp.setPerfil(new Perfil(1l, "Administrador"));
			this.usuarioPerfilCampeonatoDao.save(usuPerfilCamp);
		}
	}
	
	
}
